docs/v3/core/v3.0.rst

```diff
@@ -1211,40 +1210,42 @@ the following procedure:
-specification. In order to refer to codecs in array metadata
-documents, each codec must have a unique identifier, which is a URI
-that dereferences to a human-readable specification of the codec. A
-codec specification must declare the codec identifier, and describe
-(or cite documents that describe) the encoding and decoding algorithms
-and the format of the encoded data.
-
-A codec may have configuration parameters which modify the behaviour
-of the codec in some way. For example, a compression codec may have a
-compression level parameter, which is an integer that affects the
-resulting compression ratio of the data. Configuration parameters must
-be declared in the codec specification, including a definition of how
-configuration parameters are represented as JSON.
-
-The Zarr core development team maintains a repository of codec
-specifications, which are hosted alongside this specification in the
-`zarr-specs GitHub repository`_, and which are
-published on the `zarr-specs documentation Web site
-<https://zarr-specs.readthedocs.io/>`_. For ease of discovery, it is
-recommended that codec specifications are contributed to the
-zarr-specs GitHub repository. However, codec specifications may be
-maintained by any group or organisation and published in any location
-on the Web. For further details of the process for contributing a
-codec specification to the zarr-specs GitHub repository, see
-`ZEP 0 <https://zarr.dev/zeps/active/ZEP0000.html>`_ which describes
-the process for Zarr specification changes.
-
-Further details of how codecs are configured for an array are given in the `Array metadata`_ section.
+.. _core-codecs:
+
+Core codecs
+-----------
+
+This spec defines a set of codecs ("core codecs") which all Zarr implementations SHOULD implement in 
+order to ensure a minimal level of interoperability between Zarr implementations.
+Each core codec is each defined by specification documents which are hosted in the
+`zarr-specs GitHub repository`_, and are published on the `zarr-specs documentation web site
+<https://zarr-specs.readthedocs.io/>`_. The list of core codecs is part of the Zarr v3 specification. 
+Changes to the list of core codecs MUST be made via the same protocol used for 
+changing the Zarr v3 specification. Changes to the list of core codecs SHOULD be made carefully and
+in close collaboration with extant Zarr v3 implementations. A new core codec SHOULD be added to the 
+list when a sufficient number of Zarr implementations support or intend to support that codec. 
+An existing core codec SHOULD be removed from the list when a sufficient number of implementation 
+developers and Zarr users deem the codec worth removing, e.g. because of a technical flaw in the 
+algorithm underlying the codec.
+
+Community codecs
+----------------
+
+Zarr implementations MAY support a codec that is not in the list of core codecs 
+(hereafter termed a "community codec"), provided the community codec does not use an identifier 
```

Comment:
Is it worth namespacing codec identifiers somehow? I'm slightly against, since ideally codecs can graduate from "community" (or "extension") to "core" (after careful consideration). It'd be nice to avoid needing to rewrite metadata files when that happens.

Reply:
If we do any namespacing of the identifiers, we should not use any information about whether the codec is core or community / extension, for the reasons you note. Otherwise I don't have a preference.

Reply:
I think what is written here is fine. Namespacing codecs sounds complicated.